### PR TITLE
feat: add warnings when globals are missing

### PR DESCRIPTION
**Pull request by MatanBobi**
**What**: Adds a warning when we weren't able to setup logic through test runner globals (`afterEach`, `beforeAll`, `afterAll`).

**Why**: This was raised in https://github.com/testing-library/react-testing-library/issues/1240 and I believe it's worth adding because some stuff won't work for users so they should have a warning stating why.

**How**:

Adding `else` blocks in the relevant places.

**Checklist**:

- ~~[ ] Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs)~~
- [ ] ~~Tests~~
- [ ] ~~TypeScript definitions updated~~
- [X] Ready to be merged

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ if (typeof process === 'undefined' || !process.env?.RTL_SKIP_AUTO_CLEANUP) {
     teardown(() => {
       cleanup()
     })
+  } else {
+    console.warn(
+      `The current test runner does not support afterEach/teardown. This means we won't be able to run automatic cleanup and you should be calling cleanup() manually.`,
```
Comment (timdeschryver) [inline src/index.js:25]:
```suggestion
      `The current test runner does not support afterEach/teardown hooks. This means we won't be able to run automatic cleanup and you should be calling cleanup() manually.`,
```

File: src/index.js [hunk 2]
```diff
@@ -35,6 +39,10 @@ if (typeof process === 'undefined' || !process.env?.RTL_SKIP_AUTO_CLEANUP) {
     afterAll(() => {
       setReactActEnvironment(previousIsReactActEnvironment)
     })
+  } else {
+    console.warn(
+      'The current test runner does not support beforeAll/afterAll. This means you should be setting IS_REACT_ACT_ENVIRONMENT manually.',
```
Comment (timdeschryver) [inline src/index.js:44]:
```suggestion
      'The current test runner does not support beforeAll/afterAll hooks. This means you should be setting IS_REACT_ACT_ENVIRONMENT manually.',
```

**Review by timdeschryver — APPROVED** [review]
I like the idea!
I'm probably "stealing" this for Angular Testing Library :)

File: src/index.js (hunk 1 above)
Comment (MatanBobi) [reply src/index.js]: Thanks!

File: src/index.js (hunk 2 above)
Comment (MatanBobi) [reply src/index.js]: Thanks!

**Review by timdeschryver — APPROVED** [review]

**Comment by timdeschryver** [comment]
Thanks @ITenthusiasm that's a valid point, we'll look into this.

File: src/index.js [hunk 3]
```diff
@@ -20,6 +20,10 @@ if (typeof process === 'undefined' || !process.env?.RTL_SKIP_AUTO_CLEANUP) {
     teardown(() => {
       cleanup()
     })
+  } else {
+    console.warn(
+      `The current test runner does not support afterEach/teardown hooks. This means we won't be able to run automatic cleanup and you should be calling cleanup() manually.`,
```
Comment (eps1lon) [inline src/index.js:25]: We should add instructions on how to fix this warning i.e. import from `/pure` instead or better, ensure that you always resolve to `/pure` in your test environment.

**Pull request merged**